### PR TITLE
Add hide-file-name? option

### LINKED ISSUE #498: Option to disable source on hover?
**Is your feature request related to a problem? Please describe.**
When using the hover feature in neovim, the source displayed at the bottom is usually not very useful and adds a lot of noise:
Screen Shot 2021-07-31 at 11 03 44 AM

**Describe the solution you'd like**
It would be nice to have an option to remove the source URL at the bottom.

**Describe alternatives you've considered**
Using vim's conceal feature to hide the source URL.

**Additional context**

### PR DESCRIPTION
**Pull request by knubie**
This option hides the file name / path on hover.

Closes #498

### CODE REVIEW

File: src/clojure_lsp/feature/hover.clj [hunk 1]
```diff
@@ -34,6 +34,7 @@
 (defn hover-documentation [{sym-ns :ns sym-name :name :keys [doc filename arglist-strs] :as _definition}]
   (let [[content-format] (get-in @db/db [:client-capabilities :text-document :hover :content-format])
         show-docs-arity-on-same-line? (get-in @db/db [:settings :show-docs-arity-on-same-line?])
+        hide-filename? (get-in @db/db [:settings :hide-filename?])
```
Comment (ericdallo) [inline src/clojure_lsp/feature/hover.clj:37]: could you change to:
```suggestion
        hide-filename? (get-in @db/db [:settings :hover :hide-file-location?])
```

Comment (knubie) [reply src/clojure_lsp/feature/hover.clj:37]: How should that be reflected in `docs/settings.md`?

File: docs/settings.md [hunk 2]
```diff
@@ -46,6 +46,7 @@ This is an [example how Emacs `lsp-mode`](https://github.com/emacs-lsp/lsp-mode/
...
 | `show-docs-arity-on-same-line?`     | Whether to keep the arity on the same line of the function on hover, useful for Emacs users.                                                                                                                                                                                                                                                                                                                         | `false`                                                                                                              |
+| `hide-filename?`                    | Whether to show the full filename and path on hover.                                                                                                                                                                                                                                                                                                                                                                 | `false`                                                                                                              |
```
Comment (ericdallo) [inline docs/settings.md:49]:
```suggestion
| `hover` `hide-file-location?`                    | Whether to show the full filename and path on hover.                                                                                                                                                                                                                                                                                                                                                                 | `false`                                                                                                              |
```

File: src/clojure_lsp/feature/hover.clj (hunk 1 above)
Comment (ericdallo) [reply src/clojure_lsp/feature/hover.clj:37]: I think this is enough: https://github.com/clojure-lsp/clojure-lsp/pull/499/files#r680416292

**Comment by ericdallo** [comment]
Also, could you add a entry on CHANGELOG.md?

**Review by ericdallo — APPROVED** [review]
Looks good to me, thank you for the contribution!

**Pull request merged**